dbms/src/Functions/FunctionsArithmetic.h

```diff
@@ -1156,6 +1269,24 @@ class FunctionBinaryArithmetic : public IFunction
             return arguments[1];
         }
 
+        /// Special case - addition of two aggregate functions states
+        if (isAggregateAddition(arguments[0], arguments[1]))
+        {
+            const DataTypeAggregateFunction * new_arguments[2];
+            for (size_t i = 0; i < 2; ++i)
+                new_arguments[i] = typeid_cast<const DataTypeAggregateFunction *>(arguments[i].get());
+
+            if (!new_arguments[0]->equals(*new_arguments[1]))
+                throw Exception("Cannot add aggregate states of different functions: " 
+                    + new_arguments[0]->getFunctionName() + " and " + new_arguments[1]->getFunctionName(), ErrorCodes::CANNOT_ADD_DIFFERENT_AGGREGATE_STATES);
```

Comment:
The error message will be misleading when you try to add uniqUpTo(5) and uniqUpTo(10).